pkg/zfs/zfs_util.go

```diff
@@ -143,15 +143,21 @@ func buildCloneCreateArgs(vol *apis.ZFSVolume) []string {
 
 	if vol.Spec.VolumeType == VolTypeDataset {
 		if len(vol.Spec.Capacity) != 0 {
-			quotaProperty := "quota=" + vol.Spec.Capacity
+			quotaProperty := vol.Spec.QuotaType + "=" + vol.Spec.Capacity
 			ZFSVolArg = append(ZFSVolArg, "-o", quotaProperty)
 		}
 		if len(vol.Spec.RecordSize) != 0 {
 			recordsizeProperty := "recordsize=" + vol.Spec.RecordSize
 			ZFSVolArg = append(ZFSVolArg, "-o", recordsizeProperty)
 		}
 		if vol.Spec.ThinProvision == "no" {
-			reservationProperty := "reservation=" + vol.Spec.Capacity
+			reservationProperty := ""
+			switch vol.Spec.QuotaType {
```

Comment:
If `quota` is teh default then it should be handled in default,.

Reply:
can you explain me more? i can't get what do you mean

Reply:
@sinhaashish

Reply:
Since quota is default why to handle it as a different case. Keep it as default . 
Something like this
```
switch vol.Spec.QuotaType {
  case "refquota":
      reservationProperty = "refreservation=" + vol.Spec.Capacity
  default:
      reservationProperty = "reservation=" + vol.Spec.Capacity
}
```

Reply:
@sinhaashish I have resolved it just as you said